### DISCUSSION TITLE
Merge YAML and StructConfig nodes

### DISCUSSION THREAD

**Discussion by ireneisdoomed** [facebookresearch/hydra#2604] [Q&A]
Hello,

I have a question about what is the best way to set up a project that uses Hydra to extend a configuration both from YAMLs and from the CLI to instantiate a number of steps in my ETL.

I am using structured configs to validate my parameters, and at the moment I am not able to convey a way to extend the node that is registered in the ConfigStore from a YAML. The fact that since 1.1 Hydra requires different names for the YAML and for the class Node causes that I end up with 2 different entry points for instantiating the same class.

## Steps to reproduce
I have prepared a minimal working example to illustrate what I want to do.
```python
# cli.py
import hydra
from hydra.utils import instantiate
from omegaconf import OmegaConf

from src.config import Config, register_configs

register_configs()

@hydra.main(version_base=None, config_path="config", config_name="config")
def run_step(cfg: Config) -> None:
    print(OmegaConf.to_yaml(cfg))
    step = instantiate(cfg.step)
    step.run()


if __name__ == "__main__":
    run_step()
```

```python
# src/config.py
from dataclasses import dataclass, field
from typing import Any, List

from hydra.core.config_store import ConfigStore
from omegaconf import MISSING


@dataclass
class Config:
    """Configuration for my project. Step is a required field."""

    defaults: List[Any] = field(default_factory=lambda: [{"step": "MISSING"}])

@dataclass
class StepAConfig:
    """Config for Locus to Gene classifier."""

    input_path: str = MISSING
    output_path: str = MISSING
    id: str = "step_a"
    _target_: str = "src.steps.StepA"


def register_configs() -> None:
    cs = ConfigStore.instance()
    cs.store(name="base_config", node=Config)
    cs.store(group="step", name="base_step_a", node=StepAConfig)
```

## The problem
As I said, when I run the program I get prompted with these 2 available options, when in fact what I want to do is to feed the configuration in the step_a.yaml into the base_step_a class.

```
You must specify 'step', e.g, step=<OPTION>
Available options:
	base_step_a
	step_a

Set the environment variable HYDRA_FULL_ERROR=1 for a complete stack trace.
```

The whole project structure is here 
[my-hydra-project.zip](https://github.com/facebookresearch/hydra/files/10931505/my-hydra-project.zip)

---
I would greatly appreciate any pointers to how I could improve my setup.
Thanks!

**Comment by omry** [comment]
The Structured configs tutorial contains a [page](https://hydra.cc/docs/tutorials/structured_config/schema/) about it.
I suggest you go over the whole thing again and not start from it as it builds on the previous sections.
